### PR TITLE
ci: move coverage reporting to shared script

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by blueyed** [comment]
Interesting: the missing diff coverage between Azure/Travis is:
https://github.com/blueyed/pytest/blob/9c1bad04a109536ea69409e3c786c6761482c9c2/testing/logging/test_reporting.py#L863-L905

```
testing\logging\test_reporting.py            289    254      8      0    12%   9-24, 28-46, 50-82, 86-112, 116-132, 143-159, 170-190, 194-220, 225-253, 258-287, 292-312, 328-368, 402-443, 464-529, 541-586, 590-622, 633-669, 674-702, 707-735, 739-747, 751-786, 790-825, 829-860, 873-905, 909-918, 929-963, 967-999, 1003-1033, 1037-1081, 1089-1102, 1114-1126
testing/logging/test_reporting.py            289    229      8      0    21%   9-24, 28-46, 50-82, 86-112, 116-132, 143-159, 170-190, 194-220, 225-253, 258-287, 292-312, 328-368, 402-443, 464-529, 541-586, 590-622, 633-669, 674-702, 707-735, 739-747, 751-786, 790-825, 829-860,          909-918, 929-963, 967-999, 1003-1033, 1037-1081, 1089-1102, 1114-1126
```

**Comment by blueyed** [comment]
Thought this would be due to combining, but there is only one `.coverage` file from parallel runs, which then adds the 12%/21% (only import, no execution).

**Comment by blueyed** [comment]
Pulled out the fix for the issue also affecting Travis into https://github.com/pytest-dev/pytest/pull/5406.

**Comment by blueyed** [comment]
So this is not required anymore strictly speaking (according to https://github.com/pytest-dev/pytest/pull/5396), but I think it is good to have it in line with Travis.

Maybe we could move the common script part to an external file to keep them in sync?

**Comment by nicoddemus** [comment]
> Maybe we could move the common script part to an external file to keep them in sync?

Agreed, please do. 😁

File: azure-pipelines.yml [hunk 1]
```diff
@@ -55,8 +53,13 @@ jobs:
   - script: python -m pip install --upgrade pip && python -m pip install tox
     displayName: 'Install tox'
 
-  - script: |
-      call scripts/setup-coverage-vars.bat || goto :eof
+  - bash: |
+      if [[ "$PYTEST_COVERAGE" == "1" ]]; then
```
Comment (nicoddemus) [inline azure-pipelines.yml]: This should be calling `scripts/report-coverage.sh` no?

Comment (blueyed) [reply azure-pipelines.yml]: No, that's the env setup.

Comment (blueyed) [reply azure-pipelines.yml]: This _could_ be merged into tox.ini, and then we would end up using TOXENV=…-coverage basically.

Comment (nicoddemus) [reply azure-pipelines.yml]: Oh duh sorry

**Pull request merged**